### PR TITLE
updates

### PR DESCRIPTION
**Pull request by wsor4035**
this pull request does the following
* adds luacheck
* adds github actions with luacheck
* removes code duplication
* removes pointless dependency on default
* removes/updates depreciated api calls
* refactors the code

### CODE REVIEW

**Comment by xenonca** [comment]
I am not yet really familiar with GitHub Actions and luacheck, as soon as I've figured this out I'll check and merge this very much needed update

**Comment by wsor4035** [comment]
> I am not yet really familiar with GitHub Actions and luacheck, as soon as I've figured this out I'll check and merge this very much needed update

luacheck checks your code for accidental use of globals, line length limit, unused variables, etc. the included GitHub action simple runs luacheck on each commit and adds green checkmark or red x depending on the status 
image
you can find reports by clicking details on routine, example: https://github.com/wsor4035/ptime/runs/3265831640
image

File: README.md [hunk 1]
```diff
@@ -5,21 +5,24 @@ Makes it possible to permanently have day/night without changing the game time.
...
-default (Minetest Game)  
-unified_inventory?  
+unified_inventory
 
 
 License
 ---------
 
-Code: MIT (https://opensource.org/licenses/MIT) by xenonca  
+Code: MIT (https://opensource.org/licenses/MIT)
```
Comment (xenonca) [inline README.md]:
```suggestion
Code: [MIT](https://opensource.org/licenses/MIT) by xenonca, AnyZeeAreOkay, wsor4035
```

File: README.md [hunk 2]
```diff
@@ -5,21 +5,24 @@ Makes it possible to permanently have day/night without changing the game time.
...
+    xenonca
```
Comment (xenonca) [inline README.md]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -5,21 +5,24 @@ Makes it possible to permanently have day/night without changing the game time.
...
+    AnyZeeAreOkay
```
Comment (xenonca) [inline README.md]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -5,21 +5,24 @@ Makes it possible to permanently have day/night without changing the game time.
...
+    wsor4035
```
Comment (xenonca) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -5,21 +5,24 @@ Makes it possible to permanently have day/night without changing the game time.
...
 
 License
 ---------
 
-Code: MIT (https://opensource.org/licenses/MIT) by xenonca  
+Code: MIT (https://opensource.org/licenses/MIT)
+    xenonca
+    AnyZeeAreOkay
+    wsor4035  
 Media: MIT (https://opensource.org/licenses/MIT) by xenonca
```
Comment (xenonca) [inline README.md]:
```suggestion
Media: [MIT](https://opensource.org/licenses/MIT) by xenonca
```

File: README.md [hunk 6]
```diff
@@ -5,21 +5,24 @@ Makes it possible to permanently have day/night without changing the game time.
...
 
 
 Version
 ---------
+v1.3
```
Comment (xenonca) [inline README.md]:
```suggestion
v1.3  
```

**Review by xenonca — CHANGES_REQUESTED** [review]

**Comment by wsor4035** [comment]
@xenonca resolved

**Review by xenonca — APPROVED** [review]

**Pull request merged**